core/lib/zksync_core/src/api_server/web3/mod.rs

```diff
@@ -63,6 +65,17 @@ use self::state::{Filters, InternalApiConfig, RpcState, SealedMiniblockNumber};
 /// Timeout for graceful shutdown logic within API servers.
 const GRACEFUL_SHUTDOWN_TIMEOUT: Duration = Duration::from_secs(5);
 
+/// Represents all kinds of `Filter`.
+#[derive(Debug, Clone)]
+pub enum TypedFilter {
```

Comment:
Is `pub` visibility required here? In general, it makes sense to restrict visibility to private, then to `pub(super)` / `pub(crate)`, and only if this doesn't work, make something public. Same goes for other types you define in this PR.